src/Ocelot.Provider.Eureka/EurekaProviderFactory.cs

```diff
@@ -1,23 +1,25 @@
 ﻿using Microsoft.Extensions.DependencyInjection;
-
+using Ocelot.Configuration;
 using Ocelot.ServiceDiscovery;
-
+using Ocelot.ServiceDiscovery.Providers;
 using Steeltoe.Discovery;
+using System;
+
+namespace Ocelot.Provider.Eureka;
 
-namespace Ocelot.Provider.Eureka
+public static class EurekaProviderFactory
 {
-    public static class EurekaProviderFactory
+    public static ServiceDiscoveryFinderDelegate Get { get; } = CreateProvider;
+
+    private static IServiceDiscoveryProvider CreateProvider(IServiceProvider provider, ServiceProviderConfiguration config, DownstreamRoute route)
     {
-        public static ServiceDiscoveryFinderDelegate Get = (provider, config, route) =>
-        {
-            var client = provider.GetService<IDiscoveryClient>();
+        var client = provider.GetService<IDiscoveryClient>();
 
-            if (config.Type?.ToLower() == "eureka" && client != null)
-            {
-                return new Eureka(route.ServiceName, client);
-            }
+        if (config.Type?.ToLower() == "eureka" && client != null)
```

Comment:
Why not?
```csharp
if ("Eureka".Equals(config.Type, StringComparison.OrdinalIgnoreCase) && client != null)
{
    // ... 
}
```

Reply:
Ok, but then "Eureka" as a constant... as for PollKube.
Using a string litteral like that is imho a code smell.
I got it, you didn't like my idea of a "common" project, but it was precisely to address that kind of issues. As an example, the string litteral "servicefabric" is used several times in the solution. That's why I thought it would be nice to have a statitc class called "Constants"

Reply:
> you didn't like my idea of a "common" project

I like your idea with some Ocelot Lib project, but it must have more significant changes, not extracting these 3 constants only. 😄 
So, you have to extract all constants in all projects, create namespaces structure, make all public, write XML-docs for these common definitions...
And finally update all unit tests to remove required hardcoded constants from tests too.
It is not easy...

And why constants only?
The common Lib shoiuld have some other shared definitions ...
Well... It should be discussed and these major changes should be made in accuracy!

Reply:
> And why constants only?
> The common Lib shoiuld have some other shared definitions ...
> Well... It should be discussed and these major changes should be made in accuracy!

Yes sure, it was only an example. But it means, there is plenty to deal with. First merge the bug/poll-consul branch, then I will prepare a draft.